cmd/glasskube/cmd/bootstrap_git.go

```diff
@@ -0,0 +1,148 @@
+			cancel()
+		}
+
+		// regularly bootstrap in the cluster
+		// without a git client, this means it only supports GitHub for now
+		basePath := strings.ReplaceAll(bootstrapGitCmdOptions.url, "github.com", "raw.githubusercontent.com")
+		basePath = basePath + "/main"
+		manifestsPath := fmt.Sprintf("%v/bootstrap/glasskube/glasskube.yaml", basePath)
+		_, err := bootstrapClient.Bootstrap(ctx, bootstrap.BootstrapOptions{
+			Url:        manifestsPath,
+			Type:       bootstrap.BootstrapTypeAio,
+			GitopsMode: true,
+			Force:      true,
+		})
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "\nAn error occurred during bootstrap:\n%v\n", err)
+			cliutils.ExitWithError()
+		}
+
+		fmt.Fprintf(os.Stderr, "\nInstalling argo-cd...")
+		cliutils.SetupClientContext(true, util.Pointer(true))(cmd, make([]string, 0))
+
+		// get defined argo-cd version from repo:
+		argocdPath := fmt.Sprintf("%v/packages/argo-cd/clusterpackage.yaml", basePath)
+		argocdVersion := "v2.11.5+1"
+		if objs, err := clientutils.FetchResources(argocdPath); err != nil {
+			fmt.Fprintf(os.Stderr, "\nAn error occurred fetching argo-cd version (will use %v instead): %v\n",
+				argocdVersion, err)
+		} else {
+			for _, obj := range objs {
+				if v, ok, err :=
+					unstructured.NestedString(obj.Object, "spec", "packageInfo", "version"); !ok || err != nil {
+					fmt.Fprintf(os.Stderr, "\nAn error occurred trying to read the argo-cd version "+
+						"(will use %v instead): %v\n", argocdVersion, err)
+				} else {
+					argocdVersion = v
+					break
+				}
+			}
+		}
+		// install argo-cd package
+		argoCdPkg := client.PackageBuilder("argo-cd").
+			WithRepositoryName("glasskube").
+			WithVersion(argocdVersion).
+			BuildClusterPackage()
+		if _, err := install.NewInstaller(cliutils.PackageClient(cmd.Context())).
+			InstallBlocking(cmd.Context(), argoCdPkg, metav1.CreateOptions{}); err != nil {
+			fmt.Fprintf(os.Stderr, "\nAn error occurred installing argo-cd:\n%v\n", err)
+			cliutils.ExitWithError()
+		}
+
+		fmt.Fprintf(os.Stderr, "\nargo-cd package has been installed!\n\nApplying the glasskube argo-cd application...")
+
+		// apply bootstrap/glasskube-application.yaml into the cluster
+		appPath := fmt.Sprintf("%v/bootstrap/glasskube-application.yaml", basePath)
+		if objs, err := clientutils.FetchResources(appPath); err != nil {
+			fmt.Fprintf(os.Stderr, "\nAn error occurred fetching the bootstrap application:\n%v\n", err)
+			cliutils.ExitWithError()
+		} else if client, err := dynamic.NewForConfig(cfg); err != nil {
+			fmt.Fprintf(os.Stderr, "\nAn error occurred initializing the dynamic client:\n%v\n", err)
+			cliutils.ExitWithError()
+		} else {
+			for _, obj := range objs {
+				if _, err = client.Resource(schema.GroupVersionResource{
+					Group:    "argoproj.io",
+					Version:  "v1alpha1",
+					Resource: "applications",
+				}).Namespace(obj.GetNamespace()).Apply(ctx, obj.GetName(), &obj, metav1.ApplyOptions{
```

Comment:
using a RESTMapper would be more robust, because it ensures that the resource can be handled by the cluster and would support different kinds of resource. We do the same in pkg/bootstrap